tests/Common/BranchMetadataTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Keboola\Test\Common;
+
+use Keboola\StorageApi\BranchAwareClient;
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\DevBranches;
+use Keboola\StorageApi\DevBranchesMetadata;
+use Keboola\Test\ClientProvider\ClientProvider;
+use Keboola\Test\StorageApiTestCase;
+
+class BranchMetadataTest extends StorageApiTestCase
+{
+    const TEST_METADATA = [
+        [
+            'key' => 'KBC.SomeEnity.metadataKey',
+            'value' => 'some-value'
+        ],
+        [
+            'key' => 'someMetadataKey',
+            'value' => 'some-value'
+        ]
+    ];
+    /** @var ClientProvider */
+    private $clientProvider;
+
+    /** @var BranchAwareClient */
+    private $client;
+
+    /**
+     * @return void
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        // TODO cleanup default branch
+
+        $branches = new DevBranches($this->_client);
+        $this->deleteBranchesByPrefix($branches, $this->generateBranchNameForParallelTest());
+
+        $this->clientProvider = new ClientProvider($this);
+        $this->client = $this->clientProvider->getDefaultBranchClient();
+
+        $this->initEvents($this->client);
+    }
+
+    /**
+     * @return void
+     */
+    public function testCreateAndUpdateMetadata()
+    {
+        // create metadata client
+        $defaultMdClient = new DevBranchesMetadata($this->client);
+
+        // list metadata
+        $metadata = $defaultMdClient->listBranchMetadata();
+        // TODO change after DELETE is ready
+        //$this->assertCount(2, $metadata);
+        $this->assertIsArray($metadata);
+
+        // add metadata
+        /** @var array $metadata */
+        $metadata = $defaultMdClient->postBranchMetadata(self::TEST_METADATA);
```

Comment:
nemělo by být "addBranchMetadata"?

Reply:
IMHO v metodách máme add/remove ne post/delete

Reply:
jo, to by asi mohlo

Reply:
upraveno